app/src/main/java/com/sun/premierleague/ui/lineup/LineUpPresenter.kt

```diff
@@ -0,0 +1,134 @@
+package com.sun.premierleague.ui.lineup
+
+import com.sun.premierleague.data.model.*
+import com.sun.premierleague.data.source.FootballRepository
+import com.sun.premierleague.utils.ErrorType
+
+class LineUpPresenter(
+    private val view: LineUpContact.View,
+    private val repository: FootballRepository
+) : LineUpContact.Presenter {
+
+    override fun getLineUp(systemTeam: String, lineUp: Lineup?, isHome: Boolean) {
+        lineUp?.let {
+            val diagramTeams = mutableListOf<DiagramTeam>()
+            val systemTemps = systemTeam.trim().split(" - ")
+            val startLineups =
+                if (isHome) {
+                    lineUp.home.startingLineups as MutableList
+                } else {
+                    lineUp.away.startingLineups as MutableList
+                }.apply {
+                    sortBy { it.lineupPosition.toInt() }
+                }
+
+            if (startLineups.size >= 0) {
+                val diagrams = mutableListOf<Diagram>()
+                getDiagrams(diagrams, startLineups[0].lineupNumber, startLineups[0].lineupPlayer)
+                diagramTeams.add(DiagramTeam(diagrams))
+                startLineups.removeAt(0)
+            }
+
+            for (systemTemp in systemTemps.indices) {
+                val diagrams = mutableListOf<Diagram>()
+                val lengthSystemTemps = systemTemps[systemTemp].toInt()
+                for (item in 0 until lengthSystemTemps) {
+                    if (startLineups.size >= 0) {
+                        getDiagrams(
+                            diagrams,
+                            startLineups[0].lineupNumber,
+                            startLineups[0].lineupPlayer
+                        )
+                        startLineups.removeAt(0)
+                    }
+                }
+                diagramTeams.add(DiagramTeam(diagrams))
+            }
+            showDiagram(diagramTeams, isHome)
+        } ?: run {
+            view.showError(ErrorType.ERR_LOAD_TEAMS)
+        }
+    }
+
+    override fun getReserves(
+        substituteHomes: List<Substitute>?,
+        substituteAways: List<Substitute>?
+    ) {
+        if (substituteHomes != null && substituteAways != null) {
+            val reserves = mutableListOf<String>()
+            for (item in substituteHomes.indices) {
+                reserves.add(substituteHomes[item].lineupPlayer)
+                reserves.add(substituteAways[item].lineupPlayer)
+            }
+            view.showReserves(reserves)
+        } else {
+            view.showError(ErrorType.ERR_LOAD_RESERVE)
+        }
```

Comment:
viết theo kiểu này để giảm thiếu số lượng block {} nhé 
```suggestion
        if (substituteHomes == null && substituteAways == null) {
            view.showError(...) 
            return 
        }
```